### PR TITLE
Add Catalog and media item attribute metadata

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Ref:terraform-providers/terraform-provider-vcd#285
Added catalog item and catalog media capability to set/update/delete metadata.

```
  metadata = {
    key = "value"
    key2 = "value2"
  }
```

### CODE REVIEW

File: website/docs/r/catalog_item.html.markdown [hunk 1]
```diff
@@ -24,6 +24,12 @@ resource "vcd_catalog_item" "myNewCatalogItem" {
   ova_path             = "/home/user/file.ova"
   upload_piece_size    = 10
   show_upload_progress = true
+
+  metadata = {
+    license   = "public"    
+    version = "v1"
```
Comment (Didainius) [inline website/docs/r/catalog_item.html.markdown]: looks to be misaligned.

File: website/docs/r/catalog_media.html.markdown [hunk 2]
```diff
@@ -24,6 +24,12 @@ resource "vcd_catalog_media" "myNewMedia" {
   media_path           = "/home/user/file.iso"
   upload_piece_size    = 10
   show_upload_progress = true
+  
+  metadata = {
+    license   = "public"    
+    version = "v1"
```
Comment (Didainius) [inline website/docs/r/catalog_media.html.markdown]: Looks to be misaligned.

Comment (vbauzysvmware) [reply website/docs/r/catalog_media.html.markdown]: Added

File: website/docs/r/catalog_item.html.markdown (hunk 1 above)
Comment (vbauzysvmware) [reply website/docs/r/catalog_item.html.markdown]: Added

File: vcd/resource_vcd_catalog_item.go [hunk 3]
```diff
@@ -133,18 +140,97 @@ func resourceVcdCatalogItemCreate(d *schema.ResourceData, meta interface{}) erro
...
 }
 
 func resourceVcdCatalogItemRead(d *schema.ResourceData, meta interface{}) error {
-	return findCatalogItem(d, meta.(*VCDClient))
+	catalogItem, err := findCatalogItem(d, meta.(*VCDClient))
+	if err != nil {
+		return err
+	}
+
+	vAppTemplate, err := catalogItem.GetVAppTemplate()
```
Comment (lvirbalas) [inline vcd/resource_vcd_catalog_item.go]: Please add a comment above about why this is needed. I mean about your findings, that UI doesn't show metadata of the catalog and you need to actually set it in the child vApp template.

File: website/docs/r/catalog_item.html.markdown [hunk 4]
```diff
@@ -38,3 +44,4 @@ The following arguments are supported:
 * `ova_path` - (Required) - Absolute or relative path to file to upload
 * `upload_piece_size` - (Optional) - Size in MB for splitting upload size. It can possibly impact upload performance. Default 1MB.
 * `show_upload_progress` - (Optional) - Default false. Allows to see upload progress
+* `metadata` - (Optional; *v2.5+*) Key value map of metadata to assign to this VDC
```
Comment (lvirbalas) [inline website/docs/r/catalog_item.html.markdown]: "VDC"?

File: website/docs/r/catalog_media.html.markdown [hunk 5]
```diff
@@ -38,3 +44,4 @@ The following arguments are supported:
 * `media_path` - (Required) - Absolute or relative path to file to upload
 * `upload_piece_size` - (Optional) - size in MB for splitting upload size. It can possibly impact upload performance. Default 1MB.
 * `show_upload_progress` - (Optional) - Default false. Allows to see upload progress
+* `metadata` - (Optional; *v2.5+*) Key value map of metadata to assign to this VDC
```
Comment (lvirbalas) [inline website/docs/r/catalog_media.html.markdown]: "VDC"?

File: website/docs/r/catalog_item.html.markdown (hunk 4 above)
Comment (vbauzysvmware) [reply website/docs/r/catalog_item.html.markdown]: fixed

File: website/docs/r/catalog_media.html.markdown (hunk 5 above)
Comment (vbauzysvmware) [reply website/docs/r/catalog_media.html.markdown]: fixed

File: vcd/resource_vcd_catalog_item.go (hunk 3 above)
Comment (vbauzysvmware) [reply vcd/resource_vcd_catalog_item.go]: added

**Pull request merged**